frontend/lib/src/WidgetStateManager.ts

```diff
@@ -283,6 +283,26 @@ export class WidgetStateManager {
     }
   }
 
+  /* Sometimes users change an input field and directly click on a button to trigger a rerun
+   *  the setTimeout is used to ensure the input field value is sent to the server before the button click
+   *  by putting the the button click in the end of the event loop
+   *
+   * Returns a promise that is resolved as soon as the timeout was triggered for testing purposes.
+   */
```

Comment:
Oh, I just saw this comment as well. My understanding is that the `event loop` is the entire service that manages the non-blocking functions and the `task queue` is the queue that holds these functions in the order that they will be picked up during execution of the `event loop`. I watched this video to refresh my memory about Javascript-land after seeing your PR https://www.youtube.com/watch?v=8aGhZQkoFbQ

Reply:
Maybe if in English it said (basically a preposition distinction with English "during" versus "in"): 

```updates will be executed at the end of JavaScript's event loop```

and 

```Callbacks for other elements, for example for the onBlur event of an input field, which are already in the task queue, will then be deterministically executed before the trigger-based code during the event loop.```

or
 
```Callbacks for other elements, for example, the onBlur event of an input field, will be deterministically executed first in the event loop since they were encountered first in the sequential execution (<optional extra explanation>: and will be executed FIFO from the task queue).```

But I would argue here mentioning the queue is still a bit important, because the reason they will be executed first is because the list of tasks to execute are in a queue which is FIFO. If it were a stack, then this trick with the timeout wouldn't work.

As someone who doesn't work with Javascript all the time I needed the piece about the queue to understand the code with the 0 timeout and why this would work. Maybe our audience is more expert Javascript devs that would understand that without further details though.